webdriver/get-title.py

```diff
@@ -0,0 +1,268 @@
+import pytest
+import time
+
+from support.asserts import assert_error, assert_success
+from support.inline import inline
+
+# 1. If the current top-level browsing context is no longer open, return error
+#    with error code no such window.
+def test_title_from_closed_context(session, switch_to_inactive):
+    session.start()
+
+    switch_to_inactive()
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_error(result, "no such window")
+
+# [...]
+# 2. Handle any user prompts and return its value if it is an error.
+# [...]
+# In order to handle any user prompts a remote end must take the following
+# steps:
+# 2. Run the substeps of the first matching user prompt handler:
+#
+#    [...]
+#    - dismiss state
+#      1. Dismiss the current user prompt.
+#    [...]
+#
+# 3. Return success.
+def test_title_handle_prompt_dismiss(session, create_dialog):
+    document = inline("<title>This is the document's title</title>")
+    session.required_capabilites = {"unhandledPromptBehavior":"dismiss"}
+    session.start()
+    session.send_session_command("POST", "url", dict(url=document))
+    assert_alert_handled = create_dialog("alert")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_success(result, "This is the document's title</title>")
+    assert_alert_handled(None)
+
+    assert_confirm_handled = create_dialog("confirm")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_success(result, "This is the document's title</title>")
+    assert_confirm_handled(False)
+
+    assert_prompt_handled = create_dialog("prompt")
+
+    result = session.transport.send("GET",
+                                    "session/%s/title" % session.session_id)
+
+    assert_success(result, "This is the document's title</title>")
+    assert_prompt_handled(None)
+
+# [...]
+# 2. Handle any user prompts and return its value if it is an error.
+# [...]
+# In order to handle any user prompts a remote end must take the following
+# steps:
+# 2. Run the substeps of the first matching user prompt handler:
+#
+#    [...]
+#    - accept state
+#      1. Accept the current user prompt.
+#    [...]
+#
+# 3. Return success.
+def test_title_handle_prompt_accept(session, create_dialog):
+    document = inline("<title>This is the document's title</title>")
```

Comment:
This same document is used throughout many tests of this file, and that is fine. But could we in that case please reduce the length of it?